src/test/java/io/jenkins/plugins/pipelinegraphview/cards/items/UserIdCauseRunDetailsItemTest.java

```diff
@@ -0,0 +1,83 @@
+package io.jenkins.plugins.pipelinegraphview.cards.items;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Result;
+import hudson.model.User;
+import io.jenkins.plugins.pipelinegraphview.cards.RunDetailsItem;
+import io.jenkins.plugins.pipelinegraphview.utils.TestUtils;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Optional;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+
+@WithJenkins
+class UserIdCauseRunDetailsItemTest {
+
+    private static WorkflowRun run;
+
+    @BeforeAll
+    static void beforeAll(JenkinsRule j) throws Exception {
+        run = TestUtils.createAndRunJob(j, "simple", "singleStagePipeline.jenkinsfile", Result.SUCCESS);
+    }
+
+    @BeforeEach
+    void setUp() {
+        run.removeActions(CauseAction.class);
+        User.getAll().forEach(user -> {
+            try {
+                user.delete();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    void get_noActionsOfType() {
+        Optional<RunDetailsItem> detailsItem = UserIdCauseRunDetailsItem.get(run);
+
+        assertTrue(detailsItem.isEmpty());
+    }
+
+    @Test
+    void get_noUserIdCause() {
+        run.addAction(new CauseAction(new Cause.UpstreamCause(run)));
+
+        Optional<RunDetailsItem> detailsItem = UserIdCauseRunDetailsItem.get(run);
+
+        assertTrue(detailsItem.isEmpty());
+    }
+
+    @Test
+    void get_userNotFound() {
+        run.addAction(new CauseAction(new Cause.UserIdCause("User Id")));
+
+        Optional<RunDetailsItem> detailsItem = UserIdCauseRunDetailsItem.get(run);
+
+        assertTrue(detailsItem.isEmpty());
+    }
+
+    @Test
+    void get() {
+        run.addAction(new CauseAction(new Cause.UserIdCause("User Id")));
+        User.get("User Id", true, new HashMap<>());
+
+        Optional<RunDetailsItem> detailsItem = UserIdCauseRunDetailsItem.get(run);
+
+        assertTrue(detailsItem.isPresent());
+
+        RunDetailsItem.RunDetail userDetails = (RunDetailsItem.RunDetail) detailsItem.get();
+
+        assertEquals(new RunDetailsItem.ItemContent.PlainContent("Manually run by User Id"), userDetails.content());
```

Comment:
can you verify this passes when the locale is set to something different locally?

Reply:
I couldn't get it to pick up a locale on my machine other than english without changing the locale in the code using `Locale.setDefault` but I'm sure I am just being dumb and missing something so to account for that I've updated the assertions